devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy

```diff
@@ -107,20 +110,30 @@ def updated(){
 // Parse incoming device messages to generate events
 def parse(String description)
 {
-	def result = []
-	def cmd = zwave.parse(description, [0x31: 2, 0x30: 1, 0x84: 1])
-	if (cmd) {
-		if( cmd.CMD == "8407" ) { result << new physicalgraph.device.HubAction(zwave.wakeUpV1.wakeUpNoMoreInformation().format()) }
-		result << createEvent(zwaveEvent(cmd))
+	def results = []
+	log.debug("parsing")
+	if (description.startsWith("Err")) {
+		results = createEvent(descriptionText:description, displayed:true)
+	} else {
+		def cmd = zwave.parse(description, [0x31: 2, 0x30: 1, 0x84: 1])
+		if(cmd) results += zwaveEvent(cmd)
+		if(!results) results = [ descriptionText: cmd, displayed: false ]
 	}
-	log.debug "Parse returned ${result}"
-	return result
+	log.debug("Parsed '$description' to $results")
+	return results
 }
 
 // Event Generation
 def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd)
 {
-	[descriptionText: "${device.displayName} woke up", isStateChange: false]
+	def results = [createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)]
+	// try to configure until we've received a temperature update
+	if (!device.currentValue("temperature")) {
+		log.debug "temp is null, configuring"
+		results += configure()
```

Comment:
You'll also need to send at least some of configure() here in the case that preferences (timeout or period) have been changed since the last wakeup. See aeon-multisensor-6 for an example, where it does `setConfigured("false")` in `updated()`. `configure()` is only automatically called when the device joins, `updated()` is called whenever the preferences are changed.

Reply:
Ah, okay. I wasn't sure whether/where there was some function that was fired on preference update or not.